ruler/configs.go

```diff
@@ -0,0 +1,109 @@
+package ruler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/rules"
+)
+
+// TODO: Extract configs client logic into go client library (ala users)
+
+type cortexConfig struct {
+	RulesFiles map[string]string `json:"rules_files"`
```

Comment:
hmmm, yaml files in a json dict?  Perhaps we should just insist whatever client we write for setting this config appends all the yaml files together?